tfhe/src/integer/server_key/radix/comparison.rs

```diff
@@ -39,7 +39,93 @@ impl ServerKey {
         lhs: &RadixCiphertext<PBSOrder>,
         rhs: &RadixCiphertext<PBSOrder>,
     ) -> RadixCiphertext<PBSOrder> {
-        Comparator::new(self).unchecked_eq(lhs, rhs)
+        // Even though the corresponding function
+        // may already exist in self.key
+        // we generate our own lut do less allocations
+        let lut = self
+            .key
+            .generate_accumulator_bivariate(|x, y| u64::from(x == y));
+        let mut block_comparisons = lhs.blocks.clone();
+        block_comparisons
+            .iter_mut()
+            .zip(rhs.blocks.iter())
+            .for_each(|(lhs_block, rhs_block)| {
+                self.key
+                    .unchecked_apply_lookup_table_bivariate_assign(lhs_block, rhs_block, &lut);
+            });
+
+        let message_modulus = self.key.message_modulus.0;
+        let carry_modulus = self.key.message_modulus.0;
+        let total_modulus = message_modulus * carry_modulus;
+        let max_value = total_modulus - 1;
+
+        while block_comparisons.len() > 1 {
+            block_comparisons = block_comparisons
+                .chunks(max_value)
+                .map(|blocks| {
+                    let mut sum = blocks[0].clone();
+                    for other_block in &blocks[1..] {
+                        self.key.unchecked_add_assign(&mut sum, other_block);
+                    }
+                    let lut = self.key.generate_accumulator(|x| {
```

Comment:
Wouldn't be worth to generate these LUTs outside of the loop? The length for all but the last should be the same

Reply:
We could but to a small changed has to be made, as one the the chunk of blocks won't have the same lentgh as the others 
(we use par_chunk not par_chunk_exact)

Reply:
should be fixed, only the lut for the last chunk of blocks will be allocated